### PR TITLE
ENH: Add reporting workflow for BOLD fit

### PR DESCRIPTION
**Pull request by effigies**
## Changes proposed in this pull request

* Sets up scaffolding for including reports for fitting, following smriprep example.
* Adds basic SDC and coregistration reportlets.

TODO:

* [x] Use nireports
* [x] Add white matter contour and/or ribbon volume to show quality of SDC/coregistration

Results:

* [ds005](https://output.circle-artifacts.com/output/job/9e2d4c60-5fbb-4f2f-b129-280b60529159/artifacts/0/full-run/sub-01.html) (2 runs, no SDC)
* [ds005](https://output.circle-artifacts.com/output/job/9e2d4c60-5fbb-4f2f-b129-280b60529159/artifacts/0/partial-run/sub-01.html) (1 run, SyN-SDC)
* [ds054](https://output.circle-artifacts.com/output/job/9399b2c4-61f3-4d5e-b271-7a5861859779/artifacts/0/tmp/ds054/fmriprep/sub-100185_noerror.html) (2 run, phasediff)
* [ds210](https://output.circle-artifacts.com/output/job/b592d2a7-5cf6-445e-85e1-57bc722906b1/artifacts/0/tmp/ds210/fmriprep/sub-02.html) (1 run, SyN-SDC)

### CODE REVIEW

**Comment by effigies** [comment]
Example reports:

* ds005 ([pending](https://app.circleci.com/pipelines/github/nipreps/fmriprep/1581/workflows/bec14abd-0e7b-4d64-9992-3267f0efea11/jobs/8581))
* [ds054](https://output.circle-artifacts.com/output/job/8935910c-4afb-41d3-b061-dfb446571dbc/artifacts/0/tmp/ds054/fmriprep/sub-100185_noerror.html#datatype-figures_desc-sdc_run-01_suffix-bold_task-machinegame)
* [ds210](https://output.circle-artifacts.com/output/job/a062f5e6-3103-461e-a70f-ff58f8455fd8/artifacts/0/tmp/ds210/fmriprep/sub-02.html)

Generally we've used blue contours to indicate white matter boundaries, so I'll update that. I would also like to create a 3-way image that shows the boldref, the fieldmap reference mapped onto the boldref, and the fieldmap (in Hz) overlaid on the boldref. @oesteban Any thoughts on how to make that last happen?

**Comment by oesteban** [comment]
> I would also like to create a 3-way image that shows the boldref, the fieldmap reference mapped onto the boldref, and the fieldmap (in Hz) overlaid on the boldref. @oesteban Any thoughts on how to make that last happen?

Flickering through three frames is not possible ATM. We would need to refactor the JS code we embed in the SVGs.

However, if you don't mind the fieldmap overlay to be fixed in both (boldref and fmapref), then that should not be very difficult. And that plot makes a lot of sense to me.

WDYT?

**Comment by effigies** [comment]
Yeah, that sounds reasonable.

**Comment by effigies** [comment]
Updated the reports in the first post. This is done.

**Review by mgxd — COMMENTED** [review]
Overall reports look good 👍 

Where did the subject-specific section (# of anats, funcs, spaces, etc) of the reports go?

File: fmriprep/workflows/bold/fit.py [hunk 1]
```diff
@@ -183,12 +194,50 @@ def init_bold_fit_wf(
...
+            orientation=orientation,
+        ),
+        name="summary",
+        mem_gb=config.DEFAULT_MEMORY_MIN_GB,
+        run_without_submitting=True,
+    )
+    summary.inputs.dummy_scans = config.workflow.dummy_scans
+
+    func_fit_reports_wf = init_func_fit_reports_wf(
+        # TODO: Enable sdc report even if we find coregref
```
[the remark this answers is not in the document]
Comment (effigies) [reply fmriprep/workflows/bold/fit.py:215]: Yes.

**Comment by effigies** [comment]
> Where did the subject-specific section (# of anats, funcs, spaces, etc) of the reports go?

Made a copy of `init_single_subject_wf()` and probably didn't pull that over. Let me do that now.

File: fmriprep/workflows/bold/fit.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (effigies) [reply fmriprep/workflows/bold/fit.py:215]: Just to state this more clearly: I consider good report generation in the presence of pre-existing derivatives a second-class concern at the moment. I am okay putting out a release where that is incomplete or even containing known bugs, if that can get us to a stable state for future development and end the fork.

**Pull request merged**